uint8[] ids
uint16[] positions
uint16[] durations
